text/0000-cargo-sbom.md

```diff
@@ -0,0 +1,116 @@
+
+Adding an option to `cargo metadata` to support resolver v2 would help with overinclusion of dependencies, but still makes it difficult to ensure the exact set of features, command-line arguments, and other options are taken into account. Additionally, since build scripts (`build.rs`) may impact the output, `cargo metadata` may need to execute them.
+
+Another alternative is to extract information by setting the `RUSTC_WRAPPER` environment variable, then capture feature flags and dependencies via a wrapper tool. This would require the wrapper tool to parse the rustc command line arguments to capture the set of feature flags and referenced dependencies. This approach would prevent other uses of `RUSTC_WRAPPER`, as well as being potentially fragile.
+
+# Prior art
+[prior-art]: #prior-art
+
+* [RFC2801](https://github.com/rust-lang/rfcs/pull/2801): Proposes embedding dependency information directly into the binary. Implemented as the `cargo auditable` extension.
+* [cargo-auditable](https://github.com/rust-secure-code/cargo-auditable): Cargo extension that embeds a subset of the information described in this RFC directly into the binary. The JSON format used by this RFC could be based on the cargo-auditable format.
+* [cargo-cyclonedx](https://github.com/CycloneDX/cyclonedx-rust-cargo): Cargo extension to generate a CycloneDX SBOM.
+* [cargo-bom](https://github.com/sensorfu/cargo-bom): Cargo extension to generate a BOM in an ASCII format including license information.
+* [cargo build-plan (#5579)](https://github.com/rust-lang/cargo/issues/5579): Provides an option to emit a JSON representation of the commands to execute, without actually running them. This option has poor integration with `build.rs` and was [planned for deletion](https://github.com/rust-lang/cargo/issues/7614) in 2018.
+* [cargo unit graph (#8002)](https://github.com/rust-lang/cargo/issues/8002): Very similar to what this RFC intends on writing to disk. However, since unit-graph runs wihtout a build, it cannot take `build.rs` output into account.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+The exact specifics about what will be included in the SBOM and the specific JSON format are subject to change during the implementation of the RFC.
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+## Industry standard format
+If the software industry converges on a single, stable SBOM format, Cargo could directly emit it. The existing SBOM formats are currently changing too much at this time to standardize on a specific format.
+
+Additional fields can be added to the SBOM without a breaking change.
```

Comment:
This should perhaps be documented above, or at least mentioned alongside the need to in the future specify the format itself, that it would not be considered a breaking format change to add additional fields (more generally, the versioning / stability plan of the format is important; if it's not addressed here it ought to be addressed in the companion RFC for the format).

Reply:
I've dropped the "Industry standard format" as a future possibility.